Update Compass documentation (#835)
kyma-incubator/compass: docs/compass/03-02-labels.md

> # Labels

Reviewer: An IMO the previous version of this document was simpler :( (even if it was just a proposal for label implementation), I would suggest the following structure of the document:

>Labels allows to enrich top-level objects, like Application, Runtimes with additional information.
>
>// example, setRuntimeLabel, setApplicationLabel
>
>Label can have related Label Definition when a user can define JSON schema for validation of values.
>Label Definitions are optional, but if user use label without creating definition, LabelDefinition without any validation >rules will be created implcitly. 
>
>// example how to create Label Definition, etc
>
>Filtering collections of top-level objects is done only on the label basic.
>User has a possibility to search by label key or by label key and it's value.
>
>// examples
>
>Scenarios label
>
>// description

Also, please ensure that every example is working

Author: I guess that we both agree that the document cannot stay in the form of a proposal. Let's discuss how to improve its structure then.

A label is a key-value pair that you can add to every top-level entity, such as an Application or a Runtime. It allows you to search for and find out about all Applications and Runtimes label keys used in a given tenant. You can also define validation rules for labels with a given key using LabelDefinitions.

## LabelDefinitions

> For every label, you can create a LabelDefinition to set validation rules for values. LabelDefinitions are optional, but they are created automatically if the user adds labels for which LabelDefinitions do not exist. You can manage LabelDefinitions using mutations and queries. See the examples:

Reviewer 2: I think ` See the examples:` is not valid here, as this is our final API, not examples.

```graphql
deleteLabelDefinition(key: String!, deleteRelatedLabels: Boolean=false): LabelDefinition

```
This mutation allows you to remove only definitions that are not used. If you want to delete a LabelDefinition with all values, set the **deleteRelatedLabels** parameter to `true`.

### Search for objects using LabelFilters

You can define a LabelFilter to list the objects according to their labels. To search for a given Application or Runtime, use this query:

```graphql
 applications(filter: [LabelFilter!], first: Int = 100, after: PageCursor):  ApplicationPage!
```

To search for all objects with a given label despite their values, use this query:

```graphql
query {
  applications(filter:[{key:"scenarios"
    }]) {
    data {
      name
      labels
    }
    totalCount
  }
}
```

To search for all objects assigned to the `default` scenario, use this query:

```graphql
query {
  applications(filter:[{key:"scenarios",
    # This field is optional. If not provided, the query returns every object with the given label, regardless its value.
    query:"$[*] ? (@ == \"DEFAULT\")"}]) {
    data {
      name
      labels
    }
    totalCount
  }
}
```

> In the **query** field, use only the limited SQL/JSON path expressions. The supported syntax is `$[*] ? (@ == "{VALUE}" )`.

Reviewer 2: We should also include example for filtering by given key and value. @tgorgol has details how to do this.

## LabelFilters

You can define a LabelFilter to list the top-level entities according to their labels. You can search for Applications and Runtimes by label keys or by label keys and their values. To search for a given Application or Runtime, use this query:

```graphql
 applications(filter: [LabelFilter!], first: Int = 100, after: PageCursor):  ApplicationPage!
```

To search for all objects with a given label despite their values, use:

```graphql
query {
  applications(filter:[{key:"scenarios"
    }]) {
    data {
      name
      labels
    }
    totalCount
  }
}
```

To filter objects by their key and string value, use this query:

```graphql
runtimes(filter: { key: "{KEY}" query: "\"{VALUE}\"" })
```

You can also search for objects by their key and array values. In the **query** field, use only the limited SQL/JSON path expressions. The supported syntax is `$[*] ? (@ == "{VALUE}" )`. For example, to filter all objects assigned to the `default` scenario, run:

```graphql
query {
  applications(filter:[{key:"scenarios",
    query:"$[*] ? (@ == \"DEFAULT\")"}]) {
    data {
      name
      labels
    }
    totalCount
  }
}
```

## **Scenarios** label

> Every Application is labeled with the special **Scenarios** label which by default has the `default` value assigned. As every Application has to be assigned to at least one scenario, if not specified explicitly, the `default` scenario is used.

Reviewer 3: ```suggestion
Every Application is labeled with the special **Scenarios** label which automatically has the `default` value assigned. As every Application has to be assigned to at least one scenario, if no scenarios are explicitly specified, the `default` scenario is used.
```

> When you create a new tenant, the **Scenarios** LabelDefinition is created. It defines a list of possible values that can be used for the **Scenarios** label. Every time you create or modify an Application, there is a step that ensures that **Scenarios** label exists. You can add or remove values from the **Scenarios** LabelDefinition list, but neither the `default` value, nor the **Scenarios** label can be removed.

Reviewer 3: If you add more scenarios as values, does the `default` value remain, or do they override it?

Author: It remains.

## LabelFilters

You can define a LabelFilter to list the top-level entities according to their labels. You can search for Applications and Runtimes by label keys or by label keys and their values. To search for a given Application or Runtime, use this query:

```graphql
 applications(filter: [LabelFilter!], first: Int = 100, after: PageCursor):  ApplicationPage!
```

To search for all objects with a given label despite their values, use:

```graphql
query {
  applications(filter:[{key:"scenarios"
    }]) {
    data {
      name
      labels
    }
    totalCount
  }
}
```

To filter objects by their key and string value, use this query:

```graphql
runtimes(filter: { key: "{KEY}" query: "\"{VALUE}\"" })
```

You can also search for objects by their key and array values. In the **query** field, use only the limited SQL/JSON path expressions. The supported syntax is `$[*] ? (@ == "{VALUE}" )`. For example, to filter all objects assigned to the `default` scenario, run:

```graphql
query {
  applications(filter:[{key:"scenarios",

>     # This field is optional. If not provided, the query returns every object with the given label, regardless its value.

Reviewer: this comment duplicates information presented above in line 118.
